ambry-api/src/main/java/com/github/ambry/frontend/IdConverter.java

```diff
@@ -37,6 +39,18 @@ public interface IdConverter extends Closeable {
    */
   Future<String> convert(RestRequest restRequest, String input, Callback<String> callback);
 
+  /**
+   * Converts an ID.
+   * @param restRequest {@link RestRequest} representing the request.
+   * @param input the ID that needs to be converted.
+   * @return a {@link CompletableFuture} that will eventually contain the converted ID.
+   */
+  default CompletableFuture<String> convert(RestRequest restRequest, String input) {
```

Comment:
Looks like it is [suggested](https://stackoverflow.com/questions/47571117/what-is-the-difference-between-completionstage-and-completablefuture/49158702#49158702) to have APIs return `CompletionStage` which is an interface of `CompletableFuture` since the `future` is usually expected to be completed in the implementation and we don't expose `complete` methods to caller. Just wondering if it would be okay to return `CompletionStage` if we don't see need for caller of these APIs to complete the `future`.  (I think I may also have to do the same where I added `CompletableFutures` to other classes before).

Reply:
In principal that's true. But 1. we are already returning CompletableFuture in NamedBlobDb 2. CompletionStage has a [method](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#toCompletableFuture--) `toCompletableFuture` to return the underlying implementation. It's weird.